runtime/compiler/trj9/control/CompilationThread.cpp

```diff
@@ -6750,14 +6665,219 @@ TR::CompilationInfoPerThreadBase::compile(J9VMThread * vmThread,
+            if (entry->getMethodDetails().isDumpMethod())
+               {
+               // NOTE:
+               //       for intentional crashes, intentional traps cause the dump, and we
+               //       are not protected against them (flag: J9PORT_SIG_FLAG_SIGTRAP)
+               protectedResult = j9sig_protect((j9sig_protected_fn)wrappedCompile, static_cast<void *>(&compParam),
+                                                  (j9sig_handler_fn)  blankDumpSignalHandler, vmThread,
+                                                  flags, &result);
+               }
+            else
+               {
+               protectedResult = j9sig_protect((j9sig_protected_fn)wrappedCompile, static_cast<void*>(&compParam),
+                                                  (j9sig_handler_fn)  jitSignalHandler, vmThread,
+                                                  flags, &result);
+               }
+
+            if (protectedResult == 0) // successful completion of wrappedCompile
+               metaData = reinterpret_cast<TR_MethodMetaData *>(result);
+            else
+               metaData = 0;
+            }
+         else
+#endif
+            metaData = wrappedCompile(privatePortLibrary, &compParam);
+         }
+      else
+         {
+         entry->_compErrCode = compilationRestrictedMethod;
+         // can't compile this method: mark it appropriately
+         //if (!entry->isDLTCompile())
+         //   startPC = compilationEnd(vmThread, jitConfig, method,
+         //                            0, entry->_oldStartPC, entry->_classForNewInstance, vm);
+         }
+
+      // This method has to be called from within the try block,
+      // otherwise, the TR_Memory, TR::Region, and TR::SegmentAllocator
+      // objects go out of scope and get destroyed.
+      startPC = postCompilationTasks(vmThread, entry, method,
+                                     aotCachedMethod, metaData,
+                                     canDoRelocatableCompile, eligibleForRelocatableCompile,
+                                     reloRuntime);
+      }
+   catch (const std::exception &e)
+      {
+      entry->_compErrCode = compilationFailure;
+
+      jitConfig->totalMethodsNotTranslated++;
+
+      if (TR::Options::isAnyVerboseOptionSet(TR_VerboseCompileEnd, TR_VerboseCompFailure, TR_VerbosePerformance))
+         {
+         try
+            {
+            throw;
+            }
+         catch (const J9::JITShutdown)
+            {
+            TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,"<EARLY TRANSLATION FAILURE: JIT Shutdown signaled>");
+            }
+         catch (const std::bad_alloc &e)
+            {
+            TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,"<EARLY TRANSLATION FAILURE: out of scratch memory>");
+            }
+         catch (const std::exception &e)
+            {
+            TR_VerboseLog::writeLineLocked(TR_Vlog_FAILURE,"<EARLY TRANSLATION FAILURE: compilation aborted>");
```

Comment:
What kind of exceptions can happen here? EARLY_TRANSLATION_FAILURE was a message appearing when we couldn't allocate the "compilee" (the resolved method that needed to be compiled which now allocated in preCompilationTasks). Since this 'try' section goes around the entire compilation process it may catch more than just early failures.